src/Compilers/Core/Portable/Symbols/IMethodSymbol.cs

```diff
@@ -117,6 +117,12 @@ public interface IMethodSymbol : ISymbol
         /// </summary>
         IMethodSymbol ConstructedFrom { get; }
 
+        /// <summary>
+        /// Indicates whether the method is readonly, i.e.
+        /// whether 'this' is 'ref readonly' in the scope of the method.
+        /// </summary>
+        bool IsReadOnly { get; }
+
```

Comment:
What about an extension (`static`) method with a `this in` parameter, when viewed in reduced form?  We should have a story for how that is or is not affected.  I suggest something in the doc comment and a test.

The effect of this is more important to clients than it is to the implementation.  For example, if the method is `extern` there is no "scope of the method" but `IsReadOnly` still affects the clients.

I suggest replacing the tail of the summary with "i.e. whether the 'this' receiver parameter is 'ref readonly'" and drop mention of the scope of the method.

Reply:
Done